wire-grpc-client/src/main/java/com/squareup/wire/internal/RealGrpcStreamingCall.kt

```diff
@@ -53,9 +53,11 @@ internal class RealGrpcStreamingCall<S : Any, R : Any>(
     call.enqueue(responseChannel.readFromResponseBodyCallback(grpcMethod.responseAdapter))
 
     responseChannel.invokeOnClose {
-      call.cancel()
-      requestChannel.cancel()
-      responseChannel.cancel()
+      if (responseChannel.isClosedForReceive) {
+        // Short-circuit the request stream if it's still active.
+        call.cancel()
+        requestChannel.cancel()
+      }
```

Comment:
FYI @oldergod @Egorand I had to add the if check here too. This gets called when the stream is out of messages (not a cancel) and when the reader expresses disinterest (a cancel). We only want to break the request channel on cancels.